include/aws/common/atomic.h

```diff
@@ -0,0 +1,68 @@
+#ifndef AWS_COMMON_ATOMIC_H
+#define AWS_COMMON_ATOMIC_H
+
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/common.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Atomically retrieves the value at `dst`, and returns it.
+ */
+AWS_COMMON_API static int aws_atomic_get(int *dst);
```

Comment:
why are these static? Also it doesn't make sense to declare static linkage AND put them in the dll export.

Reply:
Static so they aren't exported symbols (to avoid duplicate symbol linkage errors). This was to address feedback I got earlier from everyone about inlining these functions. Personally I prefer not to inline stuff in general, but it did make quite a bit of sense for these atomics.